src/AppInstallerCommonCore/Public/winget/ManifestLocalization.h

```diff
@@ -3,20 +3,169 @@
 #pragma once
 #include <AppInstallerStrings.h>
 
+#include <variant>
+
 namespace AppInstaller::Manifest
 {
-    class ManifestLocalization
+    using string_t = Utility::NormalizedString;
+
+    enum class Localization : size_t
+    {
+        Publisher,
+        PublisherUrl,
+        PublisherSupportUrl,
+        PrivacyUrl,
+        Author,
+        PackageName,
+        PackageUrl,
+        License,
+        LicenseUrl,
+        Copyright,
+        CopyrightUrl,
+        ShortDescription,
+        Description,
+        Tags,
+        Max
+    };
+
+    namespace details
+    {
+        template <Localization L>
+        struct LocalizationMapping
+        {
+            // value_t type specifies the type of this data

```

Comment:
>// value_t type specifies the type of this data [](start = 12, length = 47)

Since almost everything is going to be string_t, you could define it here in the unspecialized definition.  Then one only needs to specify the very few exceptions to the rule (1 right now).